docs/assets/zdm-config-reference.yml

```diff
@@ -0,0 +1,164 @@
+# and no longer consider it as pending, thus freeing up the corresponding internal resources.
+# Note that, in this case, the ZDM Proxy will not return any result or error: when the client
+# application’s own timeout is reached, the driver will time out the request on its side.
+proxy_request_timeout_ms: 10000
+
+# Defines hot many clients may connect to single ZDM proxy instance. ZDM proxy closes
+# connection if threshold is reached.
+proxy_max_client_connections: 1000
+
+# In the CQL protocol every request has a unique id, named stream id. This variable allows
+# you to tune the maximum pool size of the available stream ids managed by the ZDM Proxy
+# per client connection. In the application client, the stream ids are managed internally
+# by the driver, and in most drivers the max number is 2048 (the same default value used
+# in the proxy). If you have a custom driver configuration with a higher value, you should
+# change this property accordingly.
+proxy_max_stream_ids: 2048
+
+# CA certificate used when verifying identity of connecting client applications.
+# proxy_tls_ca_path:
+
+# Public key used when establishing connectivity with client applications.
+# proxy_tls_cert_path:
+
+# Private key used by ZDM proxy to encrypt connection between itself and client applications
+# proxy_tls_key_path:
+
+# If true enforces mutual TLS between proxy and client applications
+# proxy_tls_require_client_auth: false
+
+# If true ZDM proxy exposes performance metrics in Prometheus format.
+metrics_enabled: true
+
+# Network interface used to expose Prometheus metrics.
+metrics_address: localhost
+
+# Port used to expose Prometheus metrics.
+metrics_port: 14001
+
+# Prefix prepended to each metric name.
+metrics_prefix: zdm
+
+# List of histogram buckets for measuring latency of origin cluster
+metrics_origin_latency_buckets_ms: 1, 4, 7, 10, 25, 40, 60, 80, 100, 150, 250, 500, 1000, 2500, 5000, 10000, 15000
+
+# List of histogram buckets for measuring latency of target cluster
+metrics_target_latency_buckets_ms: 1, 4, 7, 10, 25, 40, 60, 80, 100, 150, 250, 500, 1000, 2500, 5000, 10000, 15000
+
+# List of histogram buckets for measuring latency of asynchronous
+# read requests routed to target cluster. See parameter read_mode.
+metrics_async_read_latency_buckets_ms: 1, 4, 7, 10, 25, 40, 60, 80, 100, 150, 250, 500, 1000, 2500, 5000, 10000, 15000
+
+# Frequency (in ms) with which heartbeats will be sent on cluster connections
+# (i.e. all control and request connections to Origin and Target). Heartbeats
+# keep idle connections alive.
+heartbeat_interval_ms: 30000
+
+# Below properties define reconnection strategy for establishing control connection.
+heartbeat_retry_interval_min_ms: 250
+heartbeat_retry_interval_max_ms: 30000
+heartbeat_retry_backoff_factor: 2
+
+# Control connection failure threshold. If threshold is exceeded,
+# readiness probe of ZDM will report failure and pod will be recreated.
+heartbeat_failure_threshold: 1
```

Comment:
Maybe we can comment out all of the optional settings (but have the commented line here anyway just so that it is easy for someone to uncomment and set a specific value).

The required settings should be left uncommented like they are now 👍

Reply:
I left all required and sensitive properties uncommented, but commented majority.